hardware/XiaomiGateway.cpp

```diff
@@ -774,7 +773,7 @@ void XiaomiGateway::Do_Work()
 	XiaomiGateway::xiaomi_udp_server udp_server(io_service, m_HwdID, m_GatewayIp, m_LocalIp, m_ListenPort9898, m_OutputMessage, m_IncludeVoltage, this);
 	boost::thread bt;
 	if (m_ListenPort9898) {
-		bt = boost::thread(boost::bind(&boost::asio::io_service::run, &io_service));
+		bt = boost::thread([&io_service] { io_service.run(); });
```

Comment:
Should join this thread after calling io_service.stop() since the stop request isn't going to block.  Something like this should work after line 792:
```
if (bt.joinable())
    bt.join()
```